### ISSUE TITLE
waitSafely does not work as expected

### ISSUE THREAD

**Issue by mplogas** [jdeferred/jdeferred#51]
So, I have a small DeferredAsyncTask
```Java
public DeferredAsyncTask<Void, Integer, WebCall> getRequestTask(WebCall webcall) {
        return new DeferredAsyncTask<Void, Integer, WebCall>() {
            @Override
            protected WebCall doInBackgroundSafe(Void... voids) throws Exception {
                 //do stuff
                return webcall;
            }
        };
    }
```

that is called in a small flow
```Java
try {
    DeferredAsyncTask<Void, Integer, WebCall> requestTask = http.getRequestTask(webCall);
    new AndroidDeferredManager().when(requestTask).done(new DoneCallback<WebCall>() {
        @Override
        public void onDone(WebCall result) {
            value.set(result.getResponsePayload());
            Log.i(Constants.LOG_TAG, "webcall - result " + result.getResponsePayload());
        }
    }).fail(new FailCallback<Throwable>() {
        @Override
        public void onFail(final Throwable tr) {
            Log.i(Constants.LOG_TAG, "webcall - " + tr.getMessage());
        }
    }).waitSafely();
} catch (Exception e) {
    e.printStackTrace();
}
```
When using `waitSafely()` it just hangs. Nothing is written to the logcat, program execution is halted. Implementation seem to be fine according to the documentation. Any ideas?

**Comment by saturnism** [comment]
@mplogas which sdk version are you using?

**Comment by mplogas** [comment]
Android: SDK 21, build tools 21.1.2
Java SDK: 1.8u25

**Comment by danielfaust** [comment]
You're deadlocking the application by waiting on the UI thread, and this waiting prohibits the done or fail methods to get called on that thread. I just wonder why no ANR shows up.

**Comment by saturnism** [comment]
eak yeah.. sorry it was there mostly for testing purposes and when the deferred tasks were executing on a different thread. waitSafely should probably be removed from AndroidDeferredManager, or warn that it shouldn't be used.
